bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -2349,6 +2349,9 @@ private static boolean checkIsLikeAnydataType(Object sourceValue, Type sourceTyp
             case TypeTags.MAP_TAG:
                 return isLikeType(((MapValueImpl) sourceValue).values().toArray(), TYPE_ANYDATA,
                                   unresolvedValues, allowNumericConversion);
+            case TypeTags.TABLE_TAG:
```

Comment:
Why we need this change?

Reply:
As with these changes, the runtime `Typechecker.checkIsLikeOnValue()` will be called which comes to this `checkIsLikeAnydataType()` for the following immutable table case. We didn't have an implementation for the table cases. 
```ballerina
table<map<any>> & readonly tabRO = table [
            {id: 12, name: "abc"},
            {id: 34, name: "def"}
        ];
table<map<any>> tab = tabRO;
anydata|error result = trap <anydata>tab;
```